# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/thunderbird/sparc.patch
# Copyright (C) 2020 - 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- firefox-109.0/comm/third_party/rust/authenticator/src/transport/linux/ioctl_sparc64.rs.vanilla
+++ firefox-109.0/comm/third_party/rust/authenticator/src/transport/linux/ioctl_sparc64.rs
@@ -0,0 +1,5 @@
+/* automatically generated by rust-bindgen */
+
+pub type __u32 = ::std::os::raw::c_uint;
+pub const _HIDIOCGRDESCSIZE: __u32 = 1074022401;
+pub const _HIDIOCGRDESC: __u32 = 1342457858;
--- firefox-109.0/comm/third_party/rust/authenticator/src/transport/linux/hidwrapper.rs.vanilla	2023-01-16 18:03:41.606464537 +0100
+++ firefox-109.0/comm/third_party/rust/authenticator/src/transport/linux/hidwrapper.rs	2023-01-16 18:04:01.424465205 +0100
@@ -50,5 +50,11 @@
 #[cfg(all(target_arch = "s390x", target_endian = "big"))]
 include!("ioctl_s390xbe.rs");
 
+#[cfg(target_arch = "sparc")]
+include!("ioctl_sparc.rs");
+
+#[cfg(target_arch = "sparc64")]
+include!("ioctl_sparc64.rs");
+
 #[cfg(all(target_arch = "riscv64", target_endian = "little"))]
 include!("ioctl_riscv64.rs");
--- firefox-81.0.2/third_party/rust/authenticator/src/transport/linux/ioctl_sparc.rs.vanilla	(nonexistent)
+++ firefox-81.0.2/third_party/rust/authenticator/src/transport/linux/ioctl_sparc.rs	2020-10-16 11:38:52.534009666 +0200
@@ -0,0 +1,5 @@
+/* automatically generated by rust-bindgen */
+
+pub type __u32 = ::std::os::raw::c_uint;
+pub const _HIDIOCGRDESCSIZE: __u32 = 1074022401;
+pub const _HIDIOCGRDESC: __u32 = 1342457858;
--- firefox-109.0/third_party/rust/authenticator/build.rs.vanilla	2023-01-16 18:05:41.237468570 +0100
+++ firefox-109.0/third_party/rust/authenticator/build.rs	2023-01-16 18:06:34.859470378 +0100
@@ -47,6 +47,10 @@
         "ioctl_aarch64be.rs"
     } else if cfg!(all(target_arch = "s390x", target_endian = "big")) {
         "ioctl_s390xbe.rs"
+    } else if cfg!(target_arch = "sparc") {
+        "ioctl_sparc.rs"
+    } else if cfg!(target_arch = "sparc64") {
+        "ioctl_sparc64.rs"
     } else if cfg!(all(target_arch = "riscv64", target_endian = "little")) {
         "ioctl_riscv64.rs"
     } else {
--- firefox-122.0/third_party/rust/nix/src/sys/ioctl/linux.rs.vanilla	2024-01-28 19:13:32.067832001 +0100
+++ firefox-122.0/third_party/rust/nix/src/sys/ioctl/linux.rs	2024-01-28 19:14:20.223830081 +0100
@@ -22,6 +22,7 @@
         target_arch = "mips64",
         target_arch = "powerpc",
         target_arch = "powerpc64",
+        target_arch = "sparc",
         target_arch = "sparc64"
     ))] {
         mod consts {
